torch/csrc/distributed/c10d/ProcessGroupGloo.cpp

```diff
@@ -695,6 +695,42 @@ const std::vector<uint64_t>& ProcessGroupGloo::groupRanks() const {
   return options_->global_ranks_in_group;
 }
 
+uint64_t ProcessGroupGloo::getCommSplitCounter() const {
+  // TODO: Need to figure out a way for gloo split counter.
+  return 0;
+}
+
+c10::intrusive_ptr<Backend> ProcessGroupGloo::splitBackend(
+    const std::vector<int>& ranks,
+    const c10::intrusive_ptr<Backend::Options> opts) {
+  auto it = std::find(ranks.begin(), ranks.end(), rank_);
+  int groupRank;
+  if (it == ranks.end()) {
+    return nullptr;
+  } else {
+    groupRank = std::distance(ranks.begin(), it);
+  }
+
+  auto glooOpts = c10::dynamic_intrusive_pointer_cast<Options>(opts);
+  TORCH_CHECK(glooOpts != nullptr, "opts not a ProcessGroupGloo::Options.");
+
+  // TODO: we need to get rid of globalRanksInGroup eventually.
+  std::vector<uint64_t> globalRanksInGroup;
+  for (auto rank : ranks) {
+    globalRanksInGroup.emplace_back(groupRanks()[rank]);
+  }
+  glooOpts->global_ranks_in_group = std::move(globalRanksInGroup);
+  // TODO: Figure out a better way for split backend name.
+  glooOpts->group_name = c10::str(pg_uid_, ":split");
```

Comment:
somehow the base class does not have `group_name` attribute, unless we are ok to add one in the base class.

Reply:
doesn't hurt to add one